crates/json_parser/src/lib.rs

```diff
@@ -0,0 +1,188 @@
+use error_adapter::{Error, JsonErrorType};
+
+use tree_sitter::Node;
+use std::ops::Index;
+use std::collections::HashMap;
+
+// Enum to represent JSON values
+#[derive(Debug)]
+pub enum JsonValue {
+    Object(HashMap<String, JsonValue>),
+    Array(Vec<JsonValue>),
+    String(String),
+    Number(f64),
+    Bool(bool),
+    Null,
+}
+
+impl JsonValue {
+    pub fn as_str(&self) -> Option<&str> {
+        if let JsonValue::String(s) = self {
+            Some(s)
+        } else {
+            None
+        }
+    }
+
+    pub fn as_number(&self) -> Option<f64> {
+        if let JsonValue::Number(n) = self {
+            Some(*n)
+        } else {
+            None
+        }
+    }
+
+    pub fn as_array(&self) -> Option<&Vec<JsonValue>> {
+        if let JsonValue::Array(arr) = self {
+            Some(arr)
+        } else {
+            None
+        }
+    }
+
+    pub fn as_object(&self) -> Option<&HashMap<String, JsonValue>> {
+        if let JsonValue::Object(obj) = self {
+            Some(obj)
+        } else {
+            None
+        }
+    }
+
+    pub fn as_bool(&self) -> Option<bool> {
+        if let JsonValue::Bool(b) = self {
+            Some(*b)
+        } else {
+            None
+        }
+    }
+}
+
+// Implement indexing for JsonValue to use the [] operator
+impl Index<&str> for JsonValue {
+    type Output = JsonValue;
+
+    fn index(&self, key: &str) -> &Self::Output {
+        match self {
+            JsonValue::Object(map) => map.get(key).unwrap_or(&JsonValue::Null),
+            _ => &JsonValue::Null,
+        }
```

Comment:
Maybe `if let `will be more clear.
`index` method returns Null in two cases: 

- map has no such key 
- self is not Object.

 It may be easier to debug if this two cases are differentiated